jerry-core/lit/lit-char-helpers.c

```diff
@@ -311,6 +312,26 @@ lit_char_is_binary_digit (ecma_char_t c) /** code unit */
 } /* lit_char_is_binary_digit */
 #endif /* ENABLED (JERRY_ESNEXT) */
 
+/**
+ * UnicodeEscape abstract method
+ *
+ * See also: ECMA-262 v10, 24.5.2.3
+ */
+void
+lit_char_unicode_escape (ecma_stringbuilder_t *builder_p, /**< stringbuilder to append */
+                         ecma_char_t c) /**< code unit to convert */
+{
+  ecma_stringbuilder_append_raw (builder_p, (lit_utf8_byte_t *) "\\u", 2);
+
+  for (int8_t i = 3; i >= 0; i--)
+  {
+    int32_t result_char = (c & (((1 << 4) - 1) << (i << 2))) >> (i << 2);
```

Comment:
This looks way too overcomplicated. Wouldn't `(c >> (i * 4)) & 0xF` be enough? Or you could also do `for (int8_t i = 12; i >= 0; i -= 4)`, then you could write `(c >> i) & 0xF`.